api_shield.go

```diff
@@ -56,6 +56,8 @@ func (api *API) GetAPIShieldConfiguration(ctx context.Context, rc *ResourceConta
 func (api *API) UpdateAPIShieldConfiguration(ctx context.Context, rc *ResourceContainer, params UpdateAPIShieldParams) (Response, error) {
 	uri := fmt.Sprintf("/zones/%s/api_gateway/configuration", rc.Identifier)
 
+	params = validateAPIShieldConfigurationUpdateParams(params)
```

Comment:
I don't know how I feel about this given that an empty array for a PUT request may remove anything existing. Is there a reason we can't rely on end users to explicitly set this instead?

Reply:
Fair, i think this same approach can be done in the provider.